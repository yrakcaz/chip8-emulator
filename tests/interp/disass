0x200:	0xa2b4:		LD	I, 0x2b4
0x202:	0x23e6:		CALL	0x3e6
0x204:	0x22b6:		CALL	0x2b6
0x206:	0x7001:		ADD	V0, 0x1
0x208:	0xd011:		DRW	V0, V1, 0x1
0x20a:	0x3025:		SE	V0, 0x25
0x20c:	0x1206:		JP	0x206
0x20e:	0x71ff:		ADD	V1, 0xff
0x210:	0xd011:		DRW	V0, V1, 0x1
0x212:	0x601a:		LD	V0, 0x1a
0x214:	0xd011:		DRW	V0, V1, 0x1
0x216:	0x6025:		LD	V0, 0x25
0x218:	0x3100:		SE	V1, 0
0x21a:	0x120e:		JP	0x20e
0x21c:	0xc470:		RND	V4, 0x70
0x21e:	0x4470:		SNE	V4, 0x70
0x220:	0x121c:		JP	0x21c
0x222:	0xc303:		RND	V3, 0x3
0x224:	0x601e:		LD	V0, 0x1e
0x226:	0x6103:		LD	V1, 0x3
0x228:	0x225c:		CALL	0x25c
0x22a:	0xf515:		LD	DT, V5
0x22c:	0xd014:		DRW	V0, V1, 0x4
0x22e:	0x3f01:		SE	Vf, 0x1
0x230:	0x123c:		JP	0x23c
0x232:	0xd014:		DRW	V0, V1, 0x4
0x234:	0x71ff:		ADD	V1, 0xff
0x236:	0xd014:		DRW	V0, V1, 0x4
0x238:	0x2340:		CALL	0x340
0x23a:	0x121c:		JP	0x21c
0x23c:	0xe7a1:		SKNP	V7
0x23e:	0x2272:		CALL	0x272
0x240:	0xe8a1:		SKNP	V8
0x242:	0x2284:		CALL	0x284
0x244:	0xe9a1:		SKNP	V9
0x246:	0x2296:		CALL	0x296
0x248:	0xe29e:		SKP	V2
0x24a:	0x1250:		JP	0x250
0x24c:	0x6600:		LD	V6, 0
0x24e:	0xf615:		LD	DT, V6
0x250:	0xf607:		LD	V6, DT
0x252:	0x3600:		SE	V6, 0
0x254:	0x123c:		JP	0x23c
0x256:	0xd014:		DRW	V0, V1, 0x4
0x258:	0x7101:		ADD	V1, 0x1
0x25a:	0x122a:		JP	0x22a
0x25c:	0xa2c4:		LD	I, 0x2c4
0x25e:	0xf41e:		ADD	I, V4
0x260:	0x6600:		LD	V6, 0
0x262:	0x4301:		SNE	V3, 0x1
0x264:	0x6604:		LD	V6, 0x4
0x266:	0x4302:		SNE	V3, 0x2
0x268:	0x6608:		LD	V6, 0x8
0x26a:	0x4303:		SNE	V3, 0x3
0x26c:	0x660c:		LD	V6, 0xc
0x26e:	0xf61e:		ADD	I, V6
0x270:	0xee:		RET
0x272:	0xd014:		DRW	V0, V1, 0x4
0x274:	0x70ff:		ADD	V0, 0xff
0x276:	0x2334:		CALL	0x334
0x278:	0x3f01:		SE	Vf, 0x1
0x27a:	0xee:		RET
0x27c:	0xd014:		DRW	V0, V1, 0x4
0x27e:	0x7001:		ADD	V0, 0x1
0x280:	0x2334:		CALL	0x334
0x282:	0xee:		RET
0x284:	0xd014:		DRW	V0, V1, 0x4
0x286:	0x7001:		ADD	V0, 0x1
0x288:	0x2334:		CALL	0x334
0x28a:	0x3f01:		SE	Vf, 0x1
0x28c:	0xee:		RET
0x28e:	0xd014:		DRW	V0, V1, 0x4
0x290:	0x70ff:		ADD	V0, 0xff
0x292:	0x2334:		CALL	0x334
0x294:	0xee:		RET
0x296:	0xd014:		DRW	V0, V1, 0x4
0x298:	0x7301:		ADD	V3, 0x1
0x29a:	0x4304:		SNE	V3, 0x4
0x29c:	0x6300:		LD	V3, 0
0x29e:	0x225c:		CALL	0x25c
0x2a0:	0x2334:		CALL	0x334
0x2a2:	0x3f01:		SE	Vf, 0x1
0x2a4:	0xee:		RET
0x2a6:	0xd014:		DRW	V0, V1, 0x4
0x2a8:	0x73ff:		ADD	V3, 0xff
0x2aa:	0x43ff:		SNE	V3, 0xff
0x2ac:	0x6303:		LD	V3, 0x3
0x2ae:	0x225c:		CALL	0x25c
0x2b0:	0x2334:		CALL	0x334
0x2b2:	0xee:		RET
0x2b4:	0x8000:		LD	V0, V0
0x2b6:	0x6705:		LD	V7, 0x5
0x2b8:	0x6806:		LD	V8, 0x6
0x2ba:	0x6904:		LD	V9, 0x4
0x2bc:	0x611f:		LD	V1, 0x1f
0x2be:	0x6510:		LD	V5, 0x10
0x2c0:	0x6207:		LD	V2, 0x7
0x2c2:	0xee:		RET
0x2c4:	0x40e0:		SNE	V0, 0xe0
0x2c6:	0x0:		SYS	0
0x2c8:	0x40c0:		SNE	V0, 0xc0
0x2ca:	0x4000:		SNE	V0, 0
0x2cc:	0xe0:		CLS
0x2ce:	0x4000:		SNE	V0, 0
0x2d0:	0x4060:		SNE	V0, 0x60
0x2d2:	0x4000:		SNE	V0, 0
0x2d4:	0x4040:		SNE	V0, 0x40
0x2d6:	0x6000:		LD	V0, 0
0x2d8:	0x20e0:		CALL	0xe0
0x2da:	0x0:		SYS	0
0x2dc:	0xc040:		RND	V0, 0x40
0x2de:	0x4000:		SNE	V0, 0
0x2e0:	0xe0:		CLS
0x2e2:	0x8000:		LD	V0, V0
0x2e4:	0x4040:		SNE	V0, 0x40
0x2e6:	0xc000:		RND	V0, 0
0x2e8:	0xe0:		CLS
0x2ea:	0x2000:		CALL	0
0x2ec:	0x6040:		LD	V0, 0x40
0x2ee:	0x4000:		SNE	V0, 0
0x2f0:	0x80e0:		LD	V0, Ve
0x2f2:	0x0:		SYS	0
0x2f4:	0x40c0:		SNE	V0, 0xc0
0x2f6:	0x8000:		LD	V0, V0
0x2f8:	0xc060:		RND	V0, 0x60
0x2fa:	0x0:		SYS	0
0x2fc:	0x40c0:		SNE	V0, 0xc0
0x2fe:	0x8000:		LD	V0, V0
0x300:	0xc060:		RND	V0, 0x60
0x302:	0x0:		SYS	0
0x304:	0x80c0:		LD	V0, Vc
0x306:	0x4000:		SNE	V0, 0
0x308:	0x60:		SYS	0x60
0x30a:	0xc000:		RND	V0, 0
0x30c:	0x80c0:		LD	V0, Vc
0x30e:	0x4000:		SNE	V0, 0
0x310:	0x60:		SYS	0x60
0x312:	0xc000:		RND	V0, 0
0x314:	0xc0c0:		RND	V0, 0xc0
0x316:	0x0:		SYS	0
0x318:	0xc0c0:		RND	V0, 0xc0
0x31a:	0x0:		SYS	0
0x31c:	0xc0c0:		RND	V0, 0xc0
0x31e:	0x0:		SYS	0
0x320:	0xc0c0:		RND	V0, 0xc0
0x322:	0x0:		SYS	0
0x324:	0x4040:		SNE	V0, 0x40
0x326:	0x4040:		SNE	V0, 0x40
0x328:	0xf0:		SYS	0xf0
0x32a:	0x0:		SYS	0
0x32c:	0x4040:		SNE	V0, 0x40
0x32e:	0x4040:		SNE	V0, 0x40
0x330:	0xf0:		SYS	0xf0
0x332:	0x0:		SYS	0
0x334:	0xd014:		DRW	V0, V1, 0x4
0x336:	0x6635:		LD	V6, 0x35
0x338:	0x76ff:		ADD	V6, 0xff
0x33a:	0x3600:		SE	V6, 0
0x33c:	0x1338:		JP	0x338
0x33e:	0xee:		RET
0x340:	0xa2b4:		LD	I, 0x2b4
0x342:	0x8c10:		LD	Vc, V1
0x344:	0x3c1e:		SE	Vc, 0x1e
0x346:	0x7c01:		ADD	Vc, 0x1
0x348:	0x3c1e:		SE	Vc, 0x1e
0x34a:	0x7c01:		ADD	Vc, 0x1
0x34c:	0x3c1e:		SE	Vc, 0x1e
0x34e:	0x7c01:		ADD	Vc, 0x1
0x350:	0x235e:		CALL	0x35e
0x352:	0x4b0a:		SNE	Vb, 0xa
0x354:	0x2372:		CALL	0x372
0x356:	0x91c0:		SNE	V1, Vc
0x358:	0xee:		RET
0x35a:	0x7101:		ADD	V1, 0x1
0x35c:	0x1350:		JP	0x350
0x35e:	0x601b:		LD	V0, 0x1b
0x360:	0x6b00:		LD	Vb, 0
0x362:	0xd011:		DRW	V0, V1, 0x1
0x364:	0x3f00:		SE	Vf, 0
0x366:	0x7b01:		ADD	Vb, 0x1
0x368:	0xd011:		DRW	V0, V1, 0x1
0x36a:	0x7001:		ADD	V0, 0x1
0x36c:	0x3025:		SE	V0, 0x25
0x36e:	0x1362:		JP	0x362
0x370:	0xee:		RET
0x372:	0x601b:		LD	V0, 0x1b
0x374:	0xd011:		DRW	V0, V1, 0x1
0x376:	0x7001:		ADD	V0, 0x1
0x378:	0x3025:		SE	V0, 0x25
0x37a:	0x1374:		JP	0x374
0x37c:	0x8e10:		LD	Ve, V1
0x37e:	0x8de0:		LD	Vd, Ve
0x380:	0x7eff:		ADD	Ve, 0xff
0x382:	0x601b:		LD	V0, 0x1b
0x384:	0x6b00:		LD	Vb, 0
0x386:	0xd0e1:		DRW	V0, Ve, 0x1
0x388:	0x3f00:		SE	Vf, 0
0x38a:	0x1390:		JP	0x390
0x38c:	0xd0e1:		DRW	V0, Ve, 0x1
0x38e:	0x1394:		JP	0x394
0x390:	0xd0d1:		DRW	V0, Vd, 0x1
0x392:	0x7b01:		ADD	Vb, 0x1
0x394:	0x7001:		ADD	V0, 0x1
0x396:	0x3025:		SE	V0, 0x25
0x398:	0x1386:		JP	0x386
0x39a:	0x4b00:		SNE	Vb, 0
0x39c:	0x13a6:		JP	0x3a6
0x39e:	0x7dff:		ADD	Vd, 0xff
0x3a0:	0x7eff:		ADD	Ve, 0xff
0x3a2:	0x3d01:		SE	Vd, 0x1
0x3a4:	0x1382:		JP	0x382
0x3a6:	0x23c0:		CALL	0x3c0
0x3a8:	0x3f01:		SE	Vf, 0x1
0x3aa:	0x23c0:		CALL	0x3c0
0x3ac:	0x7a01:		ADD	Va, 0x1
0x3ae:	0x23c0:		CALL	0x3c0
0x3b0:	0x80a0:		LD	V0, Va
0x3b2:	0x6d07:		LD	Vd, 0x7
0x3b4:	0x80d2:		AND	V0, Vd
0x3b6:	0x4004:		SNE	V0, 0x4
0x3b8:	0x75fe:		ADD	V5, 0xfe
0x3ba:	0x4502:		SNE	V5, 0x2
0x3bc:	0x6504:		LD	V5, 0x4
0x3be:	0xee:		RET
0x3c0:	0xa700:		LD	I, 0x700
0x3c2:	0xf255:		LD	[I], V2
0x3c4:	0xa804:		LD	I, 0x804
0x3c6:	0xfa33:		LD	B, Va
0x3c8:	0xf265:		LD	V2, [I]
0x3ca:	0xf029:		LD	F, V0
0x3cc:	0x6d32:		LD	Vd, 0x32
0x3ce:	0x6e00:		LD	Ve, 0
0x3d0:	0xdde5:		DRW	Vd, Ve, 0x5
0x3d2:	0x7d05:		ADD	Vd, 0x5
0x3d4:	0xf129:		LD	F, V1
0x3d6:	0xdde5:		DRW	Vd, Ve, 0x5
0x3d8:	0x7d05:		ADD	Vd, 0x5
0x3da:	0xf229:		LD	F, V2
0x3dc:	0xdde5:		DRW	Vd, Ve, 0x5
0x3de:	0xa700:		LD	I, 0x700
0x3e0:	0xf265:		LD	V2, [I]
0x3e2:	0xa2b4:		LD	I, 0x2b4
0x3e4:	0xee:		RET
0x3e6:	0x6a00:		LD	Va, 0
0x3e8:	0x6019:		LD	V0, 0x19
0x3ea:	0xee:		RET
0x3ec:	0x3723:		SE	V7, 0x23
0x3ee:	0xffff:		SE	V7, 0x23
