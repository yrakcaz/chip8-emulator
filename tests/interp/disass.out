LD I, 0x2b4
CALL 0x3e6
CALL 0x2b6
ADD V0, 0x1
DRW V0, V1, 0x1
SE V0, 0x25
JP 0x206
ADD V1, 0xff
DRW V0, V1, 0x1
LD V0, 0x1a
DRW V0, V1, 0x1
LD V0, 0x25
SE V1, 0
JP 0x20e
RND V4, 0x70
SNE V4, 0x70
JP 0x21c
RND V3, 0x3
LD V0, 0x1e
LD V1, 0x3
CALL 0x25c
LD DT, V5
DRW V0, V1, 0x4
SE Vf, 0x1
JP 0x23c
DRW V0, V1, 0x4
ADD V1, 0xff
DRW V0, V1, 0x4
CALL 0x340
JP 0x21c
SKNP V7
CALL 0x272
SKNP V8
CALL 0x284
SKNP V9
CALL 0x296
SKP V2
JP 0x250
LD V6, 0
LD DT, V6
LD V6, DT
SE V6, 0
JP 0x23c
DRW V0, V1, 0x4
ADD V1, 0x1
JP 0x22a
LD I, 0x2c4
ADD I, V4
LD V6, 0
SNE V3, 0x1
LD V6, 0x4
SNE V3, 0x2
LD V6, 0x8
SNE V3, 0x3
LD V6, 0xc
ADD I, V6
RET
DRW V0, V1, 0x4
ADD V0, 0xff
CALL 0x334
SE Vf, 0x1
RET
DRW V0, V1, 0x4
ADD V0, 0x1
CALL 0x334
RET
DRW V0, V1, 0x4
ADD V0, 0x1
CALL 0x334
SE Vf, 0x1
RET
DRW V0, V1, 0x4
ADD V0, 0xff
CALL 0x334
RET
DRW V0, V1, 0x4
ADD V3, 0x1
SNE V3, 0x4
LD V3, 0
CALL 0x25c
CALL 0x334
SE Vf, 0x1
RET
DRW V0, V1, 0x4
ADD V3, 0xff
SNE V3, 0xff
LD V3, 0x3
CALL 0x25c
CALL 0x334
RET
LD V0, V0
LD V7, 0x5
LD V8, 0x6
LD V9, 0x4
LD V1, 0x1f
LD V5, 0x10
LD V2, 0x7
RET
SNE V0, 0xe0
SYS 0
SNE V0, 0xc0
SNE V0, 0
CLS
SNE V0, 0
SNE V0, 0x60
SNE V0, 0
SNE V0, 0x40
LD V0, 0
CALL 0xe0
SYS 0
RND V0, 0x40
SNE V0, 0
CLS
LD V0, V0
SNE V0, 0x40
RND V0, 0
CLS
CALL 0
LD V0, 0x40
SNE V0, 0
LD V0, Ve
SYS 0
SNE V0, 0xc0
LD V0, V0
RND V0, 0x60
SYS 0
SNE V0, 0xc0
LD V0, V0
RND V0, 0x60
SYS 0
LD V0, Vc
SNE V0, 0
SYS 0x60
RND V0, 0
LD V0, Vc
SNE V0, 0
SYS 0x60
RND V0, 0
RND V0, 0xc0
SYS 0
RND V0, 0xc0
SYS 0
RND V0, 0xc0
SYS 0
RND V0, 0xc0
SYS 0
SNE V0, 0x40
SNE V0, 0x40
SYS 0xf0
SYS 0
SNE V0, 0x40
SNE V0, 0x40
SYS 0xf0
SYS 0
DRW V0, V1, 0x4
LD V6, 0x35
ADD V6, 0xff
SE V6, 0
JP 0x338
RET
LD I, 0x2b4
LD Vc, V1
SE Vc, 0x1e
ADD Vc, 0x1
SE Vc, 0x1e
ADD Vc, 0x1
SE Vc, 0x1e
ADD Vc, 0x1
CALL 0x35e
SNE Vb, 0xa
CALL 0x372
SNE V1, Vc
RET
ADD V1, 0x1
JP 0x350
LD V0, 0x1b
LD Vb, 0
DRW V0, V1, 0x1
SE Vf, 0
ADD Vb, 0x1
DRW V0, V1, 0x1
ADD V0, 0x1
SE V0, 0x25
JP 0x362
RET
LD V0, 0x1b
DRW V0, V1, 0x1
ADD V0, 0x1
SE V0, 0x25
JP 0x374
LD Ve, V1
LD Vd, Ve
ADD Ve, 0xff
LD V0, 0x1b
LD Vb, 0
DRW V0, Ve, 0x1
SE Vf, 0
JP 0x390
DRW V0, Ve, 0x1
JP 0x394
DRW V0, Vd, 0x1
ADD Vb, 0x1
ADD V0, 0x1
SE V0, 0x25
JP 0x386
SNE Vb, 0
JP 0x3a6
ADD Vd, 0xff
ADD Ve, 0xff
SE Vd, 0x1
JP 0x382
CALL 0x3c0
SE Vf, 0x1
CALL 0x3c0
ADD Va, 0x1
CALL 0x3c0
LD V0, Va
LD Vd, 0x7
AND V0, Vd
SNE V0, 0x4
ADD V5, 0xfe
SNE V5, 0x2
LD V5, 0x4
RET
LD I, 0x700
LD [I], V2
LD I, 0x804
LD B, Va
LD V2, [I]
LD F, V0
LD Vd, 0x32
LD Ve, 0
DRW Vd, Ve, 0x5
ADD Vd, 0x5
LD F, V1
DRW Vd, Ve, 0x5
ADD Vd, 0x5
LD F, V2
DRW Vd, Ve, 0x5
LD I, 0x700
LD V2, [I]
LD I, 0x2b4
RET
LD Va, 0
LD V0, 0x19
RET
SE V7, 0x23
SE V7, 0x23
